Kernel/Ordering.hpp

```diff
@@ -69,6 +81,10 @@ class Ordering
    * under some substitutions captured by @b AppliedTerm. */
   virtual bool isGreater(AppliedTerm t1, AppliedTerm t2) const = 0;
 
+  /** Optimised function used for checking that @b lhs is greater than @b rhs,
+   * under substitution represented by @b applicator. */
+  virtual bool isGreater(TermList lhs, TermList rhs, const SubstApplicator* applicator, OrderingComparatorUP& comparator) const = 0;
```

Comment:
The implementation creates the object if needed and then calls a `const` member function. So if I'm not mistaken it would be exactly the same, except the interface would incorrectly claim that we don't modify the object. Or is it  more complicated than this?

Reply:
Ahah. Right, the *pointer* is mutable, but the object pointed-to is not (morally). Then either ignore this or consider `std::unique_ptr<const OrderingComparator>`.

Reply:
That's a good idea, it didn't occur to me first.